File: python/euler/bin/p0036.py
# project euler: problem 36

from euler.lib.util import is_palindrome


def check_palindorme(n: int) -> bool:
    return is_palindrome(n) and is_palindrome(n, base=2)


def compute(limit: int) -> str:
    return str(sum(i for i in range(1, limit, 2) if check_palindorme(i)))


def solve() -> str:
    return compute(1_000_000)

File: typescript/euler/bin/p0036.ts
// project euler: problem 36

import { isPalindrome, range } from "../lib/util.ts";

export const compute = (limit: number): string => {
  let acc = 0;
  for (const i of range(1, limit, 2)) {
    if (isPalindrome(i, 10) && isPalindrome(i, 2)) {
      acc += i;
    }
  }

  return String(acc);
};

export const solve = (): string => compute(1_000_000);
